src/main/java/studio/magemonkey/fusion/data/recipes/Recipe.java

```diff
@@ -0,0 +1,305 @@
+package studio.magemonkey.fusion.data.recipes;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.lang3.tuple.Pair;
+import org.bukkit.configuration.serialization.ConfigurationSerializable;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import studio.magemonkey.codex.util.SerializationBuilder;
+import studio.magemonkey.fusion.cfg.Cfg;
+import studio.magemonkey.fusion.cfg.hooks.divinity.DivinityRecipeMeta;
+import studio.magemonkey.fusion.data.player.FusionPlayer;
+import studio.magemonkey.fusion.data.player.PlayerLoader;
+import studio.magemonkey.fusion.data.professions.ProfessionConditions;
+import studio.magemonkey.fusion.data.professions.ProfessionResults;
+import studio.magemonkey.fusion.util.LevelFunction;
+import studio.magemonkey.fusion.util.Utils;
+import studio.magemonkey.risecore.legacy.util.DeserializationWorker;
+
+import java.util.*;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+@Getter
+@EqualsAndHashCode
+public class Recipe implements ConfigurationSerializable {
+
+    @Getter
+    private final CraftingTable table;
+
+    @Setter
+    private String name;
+
+    @Setter
+    private int craftingTime;
+    @Setter
+    private int craftingLimit;
+    @Setter
+    private int craftingLimitCooldown = -1;
+    @Setter
+    private String category;
+
+    private final ProfessionResults results;
+    private final ProfessionConditions conditions;
+
+    /* Things were recipes could be hidden */
+    @Setter
+    private Boolean hideNoPermission;
+    @Setter
+    private Boolean hideNoRank;
+    @Setter
+    private Boolean hideRecipeLimitReached;
+
+    // This is a optional meta information that is usually empty, until a recipe uses the ItemGenerator function from Divinity.
+    // I might outsource this someday but for now it's urgent and requires this solution.
+    @Getter
+    @Setter
+    private DivinityRecipeMeta divinityRecipeMeta;
```

Comment:
With a `@Getter` on the class, the other getters on these sub-fields are not necessary.

Reply:
You're right, thanks for pointing it out I forgot to clean it up.